sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -161,140 +152,52 @@ AccessToken TokenCredentialImpl::ParseToken(
-  }
+    auto const parsedJson = Azure::Core::Json::_internal::json::parse(jsonString);
 
-  {
-    auto const spacePos = expiresOn.find(' ');
-    if (spacePos != std::string::npos) // LCOV_EXCL_LINE
+    if (!parsedJson.contains(accessTokenPropertyName))
     {
-      expiresOn = expiresOn.replace(spacePos, 1, 1, 'T');
+      ThrowMissingJsonPropertyError(accessTokenPropertyName);
     }
-  }
 
-  accessToken.ExpiresOn = Azure::DateTime::Parse(expiresOn, Azure::DateTime::DateFormat::Rfc3339);
-  return accessToken;
-}
-
-namespace {
-std::string::size_type GetPropertyValueStart(
-    std::string const& jsonString,
-    std::string const& propertyName);
-
-bool GetPropertyValueAsInt64(
-    std::string const& jsonString,
-    std::string const& propertyName,
-    int64_t& outValue)
-{
-  auto const valueStartPos = GetPropertyValueStart(jsonString, propertyName);
-  if (valueStartPos == std::string::npos)
-  {
-    return false;
-  }
+    AccessToken accessToken;
+    accessToken.Token = parsedJson[accessTokenPropertyName].get<std::string>();
 
-  int64_t value = 0;
-  {
-    auto const size = jsonString.size();
-    for (auto pos = valueStartPos; pos < size; ++pos)
+    if (parsedJson.contains(expiresInPropertyName))
     {
-      auto c = jsonString[pos];
-      if (c < '0' || c > '9')
-      {
-        break;
-      }
-
-      value = (value * 10) + (static_cast<int64_t>(c) - '0');
+      accessToken.ExpiresOn
+          = std::chrono::system_clock::now()
+          + std::chrono::seconds(
+                parsedJson[expiresInPropertyName].get<std::chrono::seconds::duration::rep>());
     }
-  }
-
-  outValue = value;
-
-  return true;
-}
-
-std::string::size_type GetPropertyValueEnd(std::string const& str, std::string::size_type startPos);
-
-bool GetPropertyValueAsString(
-    std::string const& jsonString,
-    std::string const& propertyName,
-    std::string& outValue)
-{
-  auto const valueStartPos = GetPropertyValueStart(jsonString, propertyName);
-  if (valueStartPos == std::string::npos)
-  {
-    return false;
-  }
-  auto const jsonStringBegin = jsonString.begin();
-  outValue = std::string(
-      jsonStringBegin + valueStartPos,
-      jsonStringBegin + GetPropertyValueEnd(jsonString, valueStartPos));
-
-  return true;
-}
-
-std::string::size_type GetPropertyValueStart(
-    std::string const& jsonString,
-    std::string const& propertyName)
-{
-  auto const propertyNameValueSeparator = jsonString.find(':', jsonString.find(propertyName));
-  if (propertyNameValueSeparator == std::string::npos)
-  {
-    return std::string::npos;
-  }
-
-  auto pos = propertyNameValueSeparator;
-  {
-    auto const jsonStringSize = jsonString.size();
-    for (; pos < jsonStringSize; ++pos)
+    else if (expiresOnPropertyName.empty())
     {
-      auto c = jsonString[pos];
-      if (c != ':' && c != ' ' && c != '\"' && c != '\'')
-      {
-        break;
-      }
+      ThrowMissingJsonPropertyError(expiresInPropertyName);
```

Comment:
With valid response payloads from the service, will these error throw cases ever happen?